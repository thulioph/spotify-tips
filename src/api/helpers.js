import axios from 'axios';

export const requestPromise = (url) => {
  if (!url) return false;

  return new Promise((resolve, reject) => {
    axios.get(url).then(success => resolve(success)).catch(err => reject(err));
  });
};