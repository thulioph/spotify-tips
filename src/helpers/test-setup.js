import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// ====

const exposedProperties = ['window', 'navigator', 'document'];

global.window = document.defaultView;

Object.keys(document.defaultView).forEach((property) => {
    if (typeof global[property] === 'undefined') {
        exposedProperties.push(property);
        global[property] = document.defaultView[property];
    }
});

global.navigator = {
    userAgent: 'node.js'
};

// ====

Enzyme.configure({ adapter: new Adapter() });