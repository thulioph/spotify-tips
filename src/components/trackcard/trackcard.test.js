import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import chai, { expect as cExpect } from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';

import '../../helpers/test-setup';
import TrackCard from './index';

// ====

chai.use(sinonChai);

// ====

describe('<TrackCard />', () => {
    describe('Snapshot', () => {
        it('should render without crashing.', () => {
            const wrapper = shallow(<TrackCard />);
            expect(toJson(wrapper)).toMatchSnapshot();
        });
    });

    describe('Structure', () => {
        it('should have an article when mount', () => {
            const wrapper = shallow(<TrackCard />);
            cExpect(wrapper.find('article')).to.have.length(1);
        });
        
        it('should have an figure when mount', () => {
            const wrapper = shallow(<TrackCard />);
            cExpect(wrapper.find('figure')).to.have.length(1);
        });
        
        it('should have an img when mount', () => {
            const wrapper = shallow(<TrackCard />);
            cExpect(wrapper.find('img')).to.have.length(1);
        });
        
        it('should have an figcaption when mount', () => {
            const wrapper = shallow(<TrackCard />);
            cExpect(wrapper.find('figcaption')).to.have.length(1);
        });
    });
    
    describe('Props', () => {
        it('should render with the passed track image.', () => {
            const wrapper = shallow(
                <TrackCard trackImage='track-image.jpg' />
            );

            cExpect(wrapper.find('img').props().src).to.be.eql('track-image.jpg');
        });
        
        it('should render with the passed track name.', () => {
            const wrapper = shallow(
                <TrackCard trackName='I love Quebrada' />
            );

            cExpect(wrapper.find('img').props().alt).to.be.eql('I love Quebrada');
            cExpect(wrapper.find('h2').props().children).to.be.eql('I love Quebrada');
        });
        
        it('should render with the passed artist name.', () => {
            const wrapper = shallow(
                <TrackCard artistName='Emicida' />
            );

            cExpect(wrapper.find('span').props().children).to.be.eql('Emicida');
        });
    });

    describe('Events', () => {
        it('should call displayInfo when article is clicked.', () => {
            let changed = sinon.spy();

            const wrapper = mount(
                <TrackCard displayInfo={changed} />
            );
            
            wrapper.find('article').simulate('click');
            cExpect(changed).to.have.been.called;
        });
    });
});