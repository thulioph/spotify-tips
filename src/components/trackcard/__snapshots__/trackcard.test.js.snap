// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackCard /> Snapshot should render without crashing. 1`] = `
<article
  className="section-list_item"
  onClick={[Function]}
>
  <figure
    className="track-image"
  >
    <img />
    <figcaption>
      <h2 />
      <span />
    </figcaption>
  </figure>
</article>
`;
